### PR TITLE
Fix for caret rendering after focus has been lost

### LINKED ISSUE #288: Linux/Gtk: The TextBox caret is not removed after control focus is lost
Linux/Gtk: The TextBox caret is not removed after control focus is lost.

linux-caret

### PR DESCRIPTION
**Pull request by ncarrillo**
Fixes #288

### CODE REVIEW

File: src/Avalonia.Controls/Presenters/TextPresenter.cs [hunk 1]
```diff
@@ -243,7 +243,12 @@ private int CoerceCaretIndex(int value)
 
         private void CaretTimerTick(object sender, EventArgs e)
         {
-            _caretBlink = !_caretBlink;
+			if (!_caretTimer.IsEnabled) {
```
Comment (grokys) [inline src/Avalonia.Controls/Presenters/TextPresenter.cs]: Nit: can you use the same bracing style as the rest of the file?

**Comment by nc4rrillo** [comment]
@grokys Should be good to go now :-)

**Comment by grokys** [comment]
This is good as a quick fix, but I'm worried it's masking a real problem in that the tick event shouldn't be called after the timer has been disabled. Would you like to have a go at working out the source of the problem?

**Comment by nc4rrillo** [comment]
@grokys Yeah I'm going to give it a shot. I didn't realize DispatcherTimer was something that was in Avalonia. I just chalked it up to some weird framework quirk.

**Comment by grokys** [comment]
@nc4rrillo is this ready for review now?

File: src/Avalonia.Controls/Presenters/TextPresenter.cs [hunk 2]
```diff
@@ -243,7 +243,7 @@ private int CoerceCaretIndex(int value)
 
         private void CaretTimerTick(object sender, EventArgs e)
         {
-            _caretBlink = !_caretBlink;
+			_caretBlink = !_caretBlink;
```
Comment (grokys) [inline src/Avalonia.Controls/Presenters/TextPresenter.cs]: You seem to have added a tab in here.

File: src/Avalonia.Base/Threading/DispatcherTimer.cs [hunk 3]
```diff
@@ -210,7 +210,10 @@ public void Stop()
         /// </summary>
         private void InternalTick()
         {
-            Dispatcher.UIThread.InvokeAsync(_raiseTickAction, _priority);
+			if (IsEnabled)
```
Comment (grokys) [inline src/Avalonia.Base/Threading/DispatcherTimer.cs]: This still doesn't look like the source of the problem. If `IsEnabled == false` then the timer has been stopped. So why is `InternalTick` being called when the timer has stopped?

**Comment by grokys** [comment]
Closing this as it appears to me that the problem has somehow gone away. Please reopen if this isn't the case.

**Pull request closed without merging**